File: qooxdoo-contrib/QWT/branches/0.7.x/toolkit/compiler/src/test/java/org/qooxdoo/toolkit/repository/Console.java
/* ************************************************************************
   
   qooxdoo - the new era of web development
   
   http://qooxdoo.org
   
   Copyright:
     2006-2007 1&1 Internet AG, Germany, http://www.1and1.org
   
   License:
     LGPL: http://www.gnu.org/licenses/lgpl.html
     EPL: http://www.eclipse.org/org/documents/epl-v10.php
     See the LICENSE file in the project's top-level directory for details.
   
   Authors:
     * Michael Hartmeier (mlhartme)
   
 ************************************************************************ */

package org.qooxdoo.toolkit.repository;

import java.util.ArrayList;
import java.util.List;

import javax.script.ScriptException;

/** fake firebug console */
public class Console {
    public final List<String> messages;
    public final boolean firebug = true;
    
    public Console() {
        messages = new ArrayList<String>();
    }
    
    public void info(String message) throws ScriptException {
        log(message);
    }
    
    public void debug(String message) throws ScriptException {
        log(message);
    }

    public void error(String message) throws ScriptException {
        log(message);
    }

    public void log(String message) throws ScriptException {
        // System.out.println(message);
        messages.add(message);
    }

    public String last() {
        return messages.get(messages.size() - 1);
    }
}

File: trunk/qooxdoo-contrib/qcl-php/0.2/services/test/services/console.php
<?php
/* ************************************************************************

   qcl - the qooxdoo component library

   http://qooxdoo.org/contrib/project/qcl/

   Copyright:
     2007-2009 Christian Boulanger

   License:
     LGPL: http://www.gnu.org/licenses/lgpl.html
     EPL: http://www.eclipse.org/org/documents/epl-v10.php
     See the LICENSE file in the project's top-level directory for details.

   Authors:
   *  Christian Boulanger (cboulanger)

************************************************************************ */

/**
 * The url to the RpcConsole installation, relative to this file.
 * You'll need to adapt this.
 */
$consoleUrl = "../../../qooxdoo-contrib/RpcConsole/trunk/demo/default/build/";

/**
 * The url of the qcl library, relative to the url of the console url
 * You'll need to adapt this.
 */
$qclUrl = "../../../../../../qcl-php";

/**
 * The url to the backend server
 */
$serverUrl  = "$qclUrl/test/services/server.php";

/**
 * The url to the test data
 */
$testDataUrl = "$qclUrl/test/services/class/qcl/test/index.php";

/**
 * Redirect request
 */
header("Location: $consoleUrl?serverUrl=$serverUrl&testDataUrl=$testDataUrl");
?>